server/src/main/scala/org/http4s/server/middleware/Timeout.scala

```diff
@@ -33,15 +37,18 @@ object Timeout {
 
   /** Transform the service to return a RequestTimeOut [[Status]] after the supplied Duration
     * @param timeout Duration to wait before returning the RequestTimeOut
+    * @param customScheduler a custom scheduler to use for timing out the request.
+    *   If `None` is provided, then a default daemon thread scheduler will be used.
     * @param service [[HttpService]] to transform
     */
-  def apply(timeout: Duration, response: Task[Response] = DefaultTimeoutResponse)(service: HttpService): HttpService = {
-    if (timeout.isFinite()) apply(timeoutResp(timeout, response))(service)
+  def apply(timeout: Duration, response: Task[Response] = DefaultTimeoutResponse, customScheduler: Option[ScheduledExecutorService] = None)(service: HttpService): HttpService = {
+    if (timeout.isFinite()) apply(timeoutResp(timeout, response, customScheduler))(service)
     else service
   }
 
   /** Transform the service to return a RequestTimeOut [[Status]] after 30 seconds
     * @param service [[HttpService]] to transform
     */
+  @deprecated("Use the overload that also has response and customScheduler parameters (with default values provided)", "0.15")
```

Comment:
I think we could also deprecate the one on line 34.  I'm not sure what we gain by having a default timeout.